### PR TITLE
[3.11] gh-105286: Further improvements to `typing.py` docstrings (GH-105363)

### PR DESCRIPTION
**Pull request by JelleZijlstra**
(cherry picked from commit 9a89f1bf1e7bb819fe7240be779c99a84f47ea46)

* Issue: gh-105286

### CODE REVIEW

File: Lib/typing.py [hunk 1]
```diff
@@ -1736,7 +1744,7 @@ class Bar(Generic[*Ts]): ...
     Note that there is only some runtime checking of this operator. Not
     everything the runtime allows may be accepted by static type checkers.
 
-    For more information, see PEP 646.
+    For more information, see PEPs 646 and 692.
```
Comment (AlexWaygood) [inline Lib/typing.py]: PEP 692 is new in Python 3.12, so perhaps we should revert this change for the 3.11 backport:

```suggestion
    For more information, see PEP 646.
```

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
I have no idea why the "Check if generated files are up to date" check is failing (I was hoping merging in `main` would fix it) ☹️

I won't be able to merge past it, since it's a "required" CI check.

**Comment by AlexWaygood** [comment]
> Weirdly the check doesn't fail on other 3.11 PRs

It seems to be failing on all PRs that touch `.py` files, but not on PRs that touch `.rst` files. Presumably because different CI checks run on PRs that are deemed to be docs-only PRs

**Comment by AlexWaygood** [comment]
Same failures on:
- #105454
- #105445
- #105441
- #105429

**Comment by JelleZijlstra** [comment]
Opening an issue in a moment
